### PR TITLE
Fix doc files not being deleted

### PR DESCRIPTION
**Pull request by ericmj**
This should hopefully fix a bug where only unversioned doc files are being deleted when docs are republished.

This file still exists https://hexdocs.pm/phoenix/1.3.0/static_assets.html but https://hexdocs.pm/phoenix/static_assets.html doesn't, both should have been deleted.

### CODE REVIEW

File: lib/hexpm/repository/assets.ex [hunk 1]
```diff
@@ -92,36 +90,33 @@ defmodule Hexpm.Repository.Assets do
...
+    # Don't delete if we are going to overwrite with new files, this
+    # removes the downtime between a deleted and added page
+    if key in paths do
+      false
+    else
+      first = Path.relative_to(key, name) |> Path.split() |> hd()
+      case Version.parse(first) do
+        {:ok, first} ->
           # Current (/ecto/0.8.1/...)
-          first == version ->
```
Comment (ericmj) [inline lib/hexpm/repository/assets.ex]: This is the fix, the comparison was: `"1.3.0" == #Version<1.3.0>`, changed to a proper `Version.compare/2.

File: test/hexpm/web/controllers/api/docs_controller_test.exs [hunk 2]
```diff
@@ -55,42 +56,46 @@ defmodule Hexpm.Web.API.DocsControllerTest do
...
+      publish_docs(user, package, "0.0.1", [{'index.html', "package v0.0.1"}, {'remove.html', "please remove me"}])
       |> response(201)
 
       publish_docs(user, package, "0.0.1", [{'index.html', "package v0.0.1 (updated)"}])
       |> response(201)
 
       assert path("docs/#{package.name}/index.html") == "package v0.0.1 (updated)"
       assert path("docs/#{package.name}/0.0.1/index.html") == "package v0.0.1 (updated)"
+      refute path("docs/#{package.name}/remove.html")
+      refute path("docs/#{package.name}/0.0.1/remove.html")
```
Comment (wojtekmach) [inline test/hexpm/web/controllers/api/docs_controller_test.exs]: 👍

**Comment by ericmj** [comment]
@Gazler Can you try pushing phoenix 1.3.0 docs again?

**Comment by Gazler** [comment]
@ericmj Fixed! Thanks <3

**Pull request merged**